### ISSUE TITLE
[apex] apexcrudviolation not reported even if SOQL doesn't have permissions check on it

### ISSUE THREAD

**Issue by praful-gupta** [pmd/pmd#5385]
**Affects PMD Version:** 

**Rule: apexcrudviolation**

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/pmd-doc-6.55.0/pmd_rules_apex_security.html#apexcrudviolation

Even if the SOQL doesn't have any permission check on running the code analyzer for class there is no apexcrudviolation returned.

**Code Sample demonstrating the issue:**

```
 /**
 * @description  validate apex CRUD violation on SOQL
 */ 
public with sharing class CORE_PGTest{
    /**
 * @description validate apex CRUD violation on SOQL
 */ 
    public  void test(){
      List<Account> lsAcct = [select ID FROM Account ORDER BY Name ASC limit 2];

      List<Account> lsAcct2 = new List<Account>();
      lsAcct2 = [select id, Type from account limit 3];
    }   
}
```

**Expected outcome:**

PMD should report a violation at line 10, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI ]* VS Code extension for Salesforce Code Analyzer

![Uploading image.png…]
<img width="610" alt="Screenshot 2024-12-02 at 22 43 57" src="https://github.com/user-attachments/assets/ea94d350-8f17-435b-be15-9bb88981a84a">
()

**Comment by adangel** [comment]
I think, this is actually an instance of #5386 as your class ends with the suffix `Test` and the rule considers this therefore as a test class and doesn't analyze this class for valid access checks.
